### ISSUE TITLE
Gym runs endless with "Generating dSYM" inside GitHub-Actions (Question only)

### ISSUE THREAD

**Issue by tscharke** [fastlane/fastlane#15518]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject
In my [GitHub-Project](https://github.com/tscharke/fastlaneInActions/tree/features/actions) I’m trying to use GitHub-Actions as a CI to build (and later ship) a React-Native-App.

As you can see in the [Fastfile](https://github.com/tscharke/fastlaneInActions/blob/features/actions/fastlane/Fastfile) I'm using `match` to pull my certificates and `gym` to generate/build a IPA-File.

#### Question Description
The Issue I struggle with is, that the GitHub-Action cancel around 6h. And if I look closer to the logs, I see that the task: _Generating 'MyAwesomeProject.app.dSYM_ seems to run endless.

```
12:53:55]: ▸ Running script 'Bundle React Native code and images'
1130 [12:54:00]: ▸     the transform cache was reset.
1131 [12:54:23]: ▸ Generating 'MyAwesomeProject.app.dSYM'
1132 ##[error]The operation was canceled.
```

My question is a bit "general": What is the reason for this endless generation? How can I see more details? Enabling the verbose-mode give me nor more info 😞 One last idea goes through my head: Maybe it is not possible for `gym` to write to the filesystem?

It would be super if someone has an idea or see something I can not see in the moment 🙏

- [Here](https://github.com/tscharke/fastlaneInActions/commit/cca195adab1c1e99e607eb52b9ea20813f96d0f0/checks?check_suite_id=272189221#step:6:1131) is my GitHub-Action log 👉 [https://github.com/tscharke/fastlaneInActions/commit/cca195adab1c1e99e607eb52b9ea20813f96d0f0/checks?check_suite_id=272189221#step:6:1131](https://github.com/tscharke/fastlaneInActions/commit/cca195adab1c1e99e607eb52b9ea20813f96d0f0/checks?check_suite_id=272189221#step:6:1131)
- [GitHub-Action workflow](https://github.com/tscharke/fastlaneInActions/blob/features/actions/.github/workflows/ci.yml)

<details><summary>✅ fastlane environment ✅</summary>

### Stack
| Key                         | Value                                         |
| --------------------------- | --------------------------------------------- |
| OS                          | 10.14.6                                       |
| Ruby                        | 2.4.6                                         |
| Bundler?                    | true                                          |
| Git                         | git version 2.23.0                            |
| Installation Source         | ~/hostedtoolcache/Ruby/2.4.6/x64/bin/fastlane |
| Host                        | Mac OS X 10.14.6 (18G103)                     |
| Ruby Lib Dir                | ~/hostedtoolcache/Ruby/2.4.6/x64/lib          |
| OpenSSL Version             | OpenSSL 1.0.2s  28 May 2019                   |
| Is contained                | false                                         |
| Is homebrew                 | false                                         |
| Is installed via Fabric.app | false                                         |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/   |
| Xcode Version               | 10.3                                          |

### System Locale
| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:
<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#
# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

platform :ios do
  private_lane :certificates do
    match(
      app_identifier: ENV['APP_IDENTIFIER'],
      type: ENV['MATCH_TYPE'],
      readonly: true,
      shallow_clone: true,
      verbose: false,
      clone_branch_directly: true,
    )
  end

  private_lane :build do
    certificates

    gym(
      scheme: 'MyAwesomeProject',
      workspace: './ios/MyAwesomeProject.xcworkspace',
      silent: true,
      clean: true,
      # Development | Release
      configuration: 'Development',
      output_directory: './fastlane/builds/',
      output_name: 'MyAwesomeProject.ipa',
      # skip_build_archive: true,
      # skip_archive: true,
      export_options: {
          method: "ad-hoc",
      }
    )
  end

  lane :beta do
    build
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

app_identifier ENV['APP_IDENTIFIER']
apple_id ENV['APPLE_ID']
itc_team_id ENV['APPLE_TEAM_ID']

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.134.0 | ✅ Up-To-Date  |

**No plugins Loaded**
<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 2.0.2        |
| io-console                | 0.4.6        |
| CFPropertyList            | 3.0.1        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.3        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.5        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.67.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.17.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.14.1       |
| os                        | 1.0.1        |
| signet                    | 0.12.0       |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2019.1009  |
| mime-types                | 3.3          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.2.1        |
| google-cloud-core         | 1.3.2        |
| google-cloud-storage      | 1.16.0       |
| json                      | 2.2.0        |
| mini_magick               | 4.9.5        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.6        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.7.0        |
| tty-cursor                | 0.7.0        |
| tty-spinner               | 0.9.1        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.13.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2019-10-20**
</details>

**Comment by tscharke** [comment]
I really want to be honest: I need **help** (please 🙏) and I'm very sure it's no **problem** with/from Fastlane! 😉

**Comment by janpio** [comment]
Ugh, these cases where it fails inside `gym` are very hard to properly debug.

One thing you can try is to use the `disable_xcpretty` to get the plain Xcode output, that might give you some more visibility.

**Comment by tscharke** [comment]
Thanks @janpio 

> One thing you can try is to use the disable_xcpretty to get the plain Xcode output, that might give you some more visibility.
I will try it 😉

**Comment by tscharke** [comment]
With more Xcode outputs (thanks to @janpio ) I figured out what the problem was 🎉

In the end, the command `/usr/bin/codesign --force —sign …` wanna interact with the Keychain. I didn’t know the password of GitHub-Actions-Keychain. Therefor I researched in this directions and found [this](https://stackoverflow.com/questions/58118395/github-action-macos-keychain-access) at Stackoverflow.

And that's it 🙏I created my own Keychain when I'm on the CI-system and use it.
